datadog_sync/model/monitors.py

```diff
@@ -4,99 +4,74 @@
+from typing import Optional, List, Dict
 
-from requests.exceptions import HTTPError
-
-from datadog_sync.utils.base_resource import BaseResource
+from datadog_sync.utils.base_resource import BaseResource, ResourceConfig
+from datadog_sync.utils.custom_client import CustomClient
 from datadog_sync.utils.resource_utils import ResourceConnectionError
 
 
 class Monitors(BaseResource):
     resource_type = "monitors"
-    resource_connections = {"monitors": ["query"], "roles": ["restricted_roles"]}
-    base_path = "/api/v1/monitor"
-    excluded_attributes = [
-        "root['id']",
-        "root['matching_downtimes']",
-        "root['creator']",
-        "root['created']",
-        "root['deleted']",
-        "root['org_id']",
-        "root['created_at']",
-        "root['modified']",
-        "root['overall_state']",
-        "root['overall_state_modified']",
-    ]
-
-    def import_resources(self):
-        source_client = self.config.source_client
-        try:
-            resp = source_client.get(self.base_path).json()
-        except HTTPError as e:
-            self.logger.error("error importing monitors %s", e)
+    resource_config = ResourceConfig(
+        resource_connections={"monitors": ["query"], "roles": ["restricted_roles"]},
+        base_path="/api/v1/monitor",
+        excluded_attributes=[
+            "id",
+            "matching_downtimes",
+            "creator",
+            "created",
+            "deleted",
+            "org_id",
+            "created_at",
+            "modified",
+            "overall_state",
+            "overall_state_modified",
+        ],
+    )
+    # Additional Monitors specific attributes
+
+    def get_resources(self, client: CustomClient) -> List[Dict]:
+        resp = client.get(self.resource_config.base_path).json()
+
+        return resp
+
+    def import_resource(self, resource: Dict) -> None:
+        if resource["type"] == "synthetics alert":
             return
 
-        self.import_resources_concurrently(resp)
-
-    def process_resource_import(self, monitor):
-        if not self.filter(monitor) or monitor["type"] == "synthetics alert":
-            return
+        self.resource_config.source_resources[str(resource["id"])] = resource
 
-        self.source_resources[str(monitor["id"])] = monitor
+    def pre_resource_action_hook(self, resource: Dict) -> None:
+        pass
 
-    def apply_resources(self):
+    def pre_apply_hook(self, resources: Dict[str, Dict]) -> Optional[list]:
         simple_monitors = {}
         composite_monitors = {}
 
-        for _id, monitor in self.source_resources.items():
-            if monitor["type"] == "synthetics alert":
-                continue
+        for _id, monitor in self.resource_config.source_resources.items():
             if monitor["type"] == "composite":
                 composite_monitors[_id] = monitor
             else:
                 simple_monitors[_id] = monitor
+        return [simple_monitors, composite_monitors]
```

Comment:
```suggestion
        return simple_monitors + composite_monitors
```
We can merge them and keep the list flat, right?

Reply:
No we cannot. Combining the list would mean that there is a possibility that composite monitor is applied at the same time as simple monitors.

This goes back to your original approach as well. Pretty much the same concept but without having to call apply manually twice.